ludwig/benchmarking/artifacts.py

```diff
@@ -0,0 +1,25 @@
+import os
+
+from ludwig.utils.data_utils import load_json, load_yaml
+
+
+class BenchmarkingArtifact:
```

Comment:
I like this idea of defining a compositional return type that users can subsequently interface with after calling `benchmark()`.

`ludwig experiment` and `ludwig train` should move to this pattern as well, if/when we move all of the modeling artifacts move to dataclasses, CC: @dantreiman 

It looks like, besides the `load_yaml` unpacking, this could be as simple `dataclass`.

Suggestion: define a dataclass, i.e. `BenchmarkingResult`, and a method that creates that dataclass, i.e. `build_benchmarking_results(benchmarking_config: dict, experiment_idx: int)`.

Then the return of `benchmark` could be a dict of experiment name (or other id) (str) -> `BenchmarkingResult`.